relay/pkg/chain/chain.go

```diff
@@ -10,44 +10,45 @@ type Handle interface {
 // Relay is an interface that provides ability to interact with Relay contract.
 type Relay interface {
 	// GetBestKnownDigest returns the best known digest.
-	GetBestKnownDigest() ([32]uint8, error)
+	GetBestKnownDigest() ([32]byte, error)
```

Comment:
Should we declare `type Digest [32]byte` just like we do for BTC? There is a lot of `[32]byte` declarations in this file.

Reply:
I thought about this but this will require conversions between `btc.Digest` and `chain.Digest` in the `block.Forwarder`. That seemed odd to me as `block.Forwarder` shouldn't care about data conversions but just forward headers according to defined logic. Of course, this is a simple conversion but still, I'm not 100% sure.

Reply:
I think this conversion feels wrong because we would be converting from BTC header to... BTC header but declared in the host chain package. Instead of having two (identical) representations of BTC header, maybe the interface from `pkg/chain/chain.go` should import types from `pkg/btc/btc.go`?

Reply:
Yeah, importing types from `btc.go` makes sense. However I will do this in a follow-up PR once the whole stack is merged. Left a TODO for now: https://github.com/keep-network/tbtc/commit/624b6750d90127716326f460c7c934a16f9125de

Reply:
Yeah, separate PR makes sense :+1: